src/game.py

```diff
@@ -0,0 +1,227 @@
+        for position in order:
+            if deduced[position] is None:
+                puzzle.append((position, board[position]))
+                deduced[position] = board[position]
+                self.deduce(deduced)
+        return self.board_for_entries(puzzle)
+
+    def board_for_entries(self, entries):
+        """Get array with tuples of position and value.
+        This method will return an array with final array
+        with values (Numbers and None).
+
+        Keyword arguments:
+        entries -- Array of tuples e.g.: [(23, 5), (80, 9),..., (2, 5), (15, 1)]
+        return  -- Array e.g.: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        """
+        board = [None] * 81
+        for position, number in entries:
+            board[position] = number
+        return board
+
+    def solution(self, board):
+        """Get an empty array with 81 positions and call
+        solve_board method to get a solved sudoku.
+
+        Keyword arguments:
+        board -- Array e.g.: [None, None,..., None, None]
+        return  -- Array e.g.: [4, 9, 3, 4, 5, ...., 8, 7, 4, 8, 1]
+        """
+        return self.solve_board(board)[1]
+
+    def solve_board(self, original):
+        """Get an empty array (it can be resolve the puzzle)
+        and create an arrays of board in order to get an unique solution
+
+        Keyword arguments:
+        original -- Array of Arrays e.g.: [[None, None,..., None, None]]
+        return  -- Array e.g.: [4, 9, 3, 4, 5, ...., 8, 7, 4, 8, 1]
+        """
+        board = list(original)
+        guesses = self.deduce(board)
+        if guesses is None:
+            return [], board
+        track = [(guesses, 0, board)]
+        return self.solve_next(track)
+
+    def solve_next(self, remembered):
+        """Get a board and guesses required if game is empty,
+        it will create all game until a board does not contains None values
+
+        Keyword arguments:
+        remembered -- Array e.g.: [[None, None,..., None, None]]
+        return  -- Array e.g.: [4, 9, 3, 4, 5, ...., 8, 7, 4, 8, 1]
+        """
+        while len(remembered) > 0:
+            guesses, count, board = remembered.pop()
+            if count >= len(guesses):
+                continue
+            remembered.append((guesses, count + 1, board))
+            workspace = list(board)
+            pos, n = guesses[count]
+            workspace[pos] = n
+            guesses = self.deduce(workspace)
+            if guesses is None:
+                return (remembered, workspace)
+            remembered.append((guesses, 0, workspace))
+        return [], None
+
+    def deduce(self, board):
+        """Get a board that require to deduce the value following the
+         sudoku rules
+
+        Keyword arguments:
+        board -- Array e.g.: [[None, None,..., None, None]]
+        return  -- Array of tuples [(Position, Value)], the number of array items depends of number gasses found
+        """
+        while True:
```

Comment:
Method with more than 20 lines of code and more than 2 nested levels, need to be refactored